x-pack/test/api_integration/deployment_agnostic/apis/observability/streams/classic_forking.ts

```diff
@@ -325,5 +327,59 @@ export default function ({ getService }: DeploymentAgnosticFtrProviderContext) {
         400
       );
     });
+
+    it('should attach to the correct unwired stream in case of matching prefix', async () => {
+      await esClient.indices.createDataStream({ name: 'logs-shared-default' });
+      await esClient.indices.createDataStream({ name: 'logs-shared.prefix-default' });
+
+      await putStream(apiClient, 'logs-shared.prefix.myfork-default', {
```

Comment:
What happens to `logs-shared.prefix-default` if we first fork on `logs-shared-default` instead?

Reply:
It just sits there as an unwired stream, e.g. you could have unwired `logs-shared-default` with a wired child `logs-shared.myfork-default` and unwired `logs-shared.prefix-default` with a wired child `logs-shared.prefix.myfork-default`.

If a fork would cause a conflict, it responds a 400.

It's a good case to validate though, I added an integration test for it.